Sources/Restler/Public/Classes/Request/RequestBuilder.swift

```diff
@@ -4,86 +4,74 @@ typealias QueryParametersType = [URLQueryItem]
-        private let multipartEncoder: RestlerMultipartEncoderType
-        private let dispatchQueueManager: DispatchQueueManagerType
-        private let errorParser: RestlerErrorParserType
-        private let method: HTTPMethod
-        private let endpoint: RestlerEndpointable
-        
-        private var header: Restler.Header
-        private var query: QueryParametersType?
-        private var body: Data?
-        private var errors: [Error] = []
+        private let dependencies: Dependencies
+        private var form: Form
         
         // MARK: - Initialization
         internal init(
-            baseURL: URL,
-            networking: NetworkingType,
-            encoder: RestlerJSONEncoderType,
-            decoder: RestlerJSONDecoderType,
-            queryEncoder: RestlerQueryEncoderType,
-            multipartEncoder: RestlerMultipartEncoderType,
-            dispatchQueueManager: DispatchQueueManagerType,
-            errorParser: RestlerErrorParserType,
-            method: HTTPMethod,
-            endpoint: RestlerEndpointable,
+            dependencies: Dependencies,
             header: Restler.Header
         ) {
-            self.baseURL = baseURL
-            self.networking = networking
-            self.encoder = encoder
-            self.decoder = decoder
-            self.queryEncoder = queryEncoder
-            self.multipartEncoder = multipartEncoder
-            self.dispatchQueueManager = dispatchQueueManager
-            self.errorParser = errorParser
-            self.method = method
-            self.endpoint = endpoint
-            self.header = header
+            self.dependencies = dependencies
+            self.form = Form(header: header)
         }
     }
 }
 
+// MARK: - Structures
+extension Restler.RequestBuilder {
+    struct Dependencies {
+        let url: URL
+        let networking: NetworkingType
+        let encoder: RestlerJSONEncoderType
+        let decoder: RestlerJSONDecoderType
+        let queryEncoder: RestlerQueryEncoderType
+        let multipartEncoder: RestlerMultipartEncoderType
+        let dispatchQueueManager: DispatchQueueManagerType
+        let errorParser: RestlerErrorParserType
+        let method: HTTPMethod
+    }
+    
+    struct Form {
+        var header: Restler.Header
+        var query: QueryParametersType?
+        var body: Data?
+        var errors: [Restler.Error] = []
+        var customRequestModification: ((inout URLRequest) -> Void)?
+    }
+}
+
 // MARK: - RestlerBasicRequestBuilderType
 extension Restler.RequestBuilder: RestlerBasicRequestBuilderType {
     public func setInHeader(_ value: String?, forKey key: Restler.Header.Key) -> Self {
-        self.header[key] = value
+        self.form.header[key] = value
         return self
     }
     
     public func failureDecode<T>(_ type: T.Type) -> Self where T: RestlerErrorDecodable {
-        self.errorParser.decode(type)
+        self.dependencies.errorParser.decode(type)
```

Comment:
ErrorParser isn't copied here, so it will use provided here type globally. We should either copy it or make a structure instead of a class